### DISCUSSION TITLE
question about linked library of linuxbrew's executable file

### DISCUSSION THREAD

**Discussion by hei-threebody** [Homebrew/discussions#99] [Everyday usage] [answered]
I am wondering why the dynamic library of linuxbrew's installed executable file will not change with the LD_LIBRARY_PATH. There are some command line tools in my system that will broken when the old lidstdc++ library is added to LD_LIBRARY_PATH. However, the tools installed with linuxbrew are never broken.

I look into the rpath which chrpath and patchelf, but cannot make my broken command line tools work as the linuxbrew's ones. Dose any one know which magic linuxbrew is using?

**Comment by SMillerDev** [comment]
@Homebrew/linux will know this

**Comment by sjackman** [comment, marked as answer]
`RPATH` is searched before `LD_LIBRARY_PATH` is searched before `RUNPATH`.
See https://amir.rachum.com/blog/2016/09/17/shared-libraries/
Homebrew on Linux uses `patchelf --force-rpath --set-rpath=…` to use `RPATH` rather than the default `RUNPATH`.
Note that `RPATH` predates the creation of `RUNPATH`.
